src/modules/job-manager/queue.c

```diff
@@ -311,16 +311,45 @@ int queue_submit_check (struct queue *queue,
     return 0;
 }
 
-bool queue_started (struct queue *queue)
+static int job_queue_name (struct queue *queue, struct job *job, const char **qnamep)
+{
+    json_error_t error;
+    const char *qname = NULL;
+
+    if (!job || !job->jobspec_redacted) {
+        errno = EINVAL;
+        return -1;
+    }
+
+    if (json_unpack_ex (job->jobspec_redacted, &error, 0,
```

Comment:
It might be more efficient to parse this on job creation into a `job->queue` member.  Then you could follow the existing error handling of the "create"  functions if something is wrong
- if job is created from eventlog, propagate the json parse error (which is ignored here)
- if job is created from json, return generic failure since jobspec was already validated at ingest.

And this parsing step would be done once not possibly multiple times per job.

Reply:
can the `jobspec-update` event be used to alter a job's queue?  Right now, it appears the answer is yes, so would have to handle this case.

Reply:
No, we don't currently support that.